embulk-api/src/main/java/org/embulk/config/ConfigSource.java

```diff
@@ -31,11 +31,17 @@
     /**
      * Loads this configuration as a task class like {@code PluginTask}.
      *
+     * <p>This method will be deprecated. Loading configs will be replaced by another library
```

Comment:
Nit: Having `TODO` or `@Deprecated` (probably not what you want, since don't deprecate now) would make it easier to find it later.

Reply:
Yeah, but it's not deprecated yet as you said. The purpose of this comment is to describe it "will be" deprecated.

Also, this is published Javadoc, not our internal comments. TODO may not fit for Javadoc.